src/celeritas/field/MagFieldEquation.hh

```diff
@@ -90,18 +90,25 @@ template<class FieldT>
 CELER_FUNCTION auto
 MagFieldEquation<FieldT>::operator()(const OdeState& y) const -> OdeState
 {
-    // Get a magnetic field value at a given position
-    auto&& mag_vec = calc_field_(y.pos);
-
-    real_type momentum_mag2 = dot_product(y.mom, y.mom);
-    CELER_ASSERT(momentum_mag2 > 0);
-    real_type momentum_inv = 1 / std::sqrt(momentum_mag2);
+    CELER_EXPECT(y.mom[0] != 0 || y.mom[1] != 0 || y.mom[2] != 0);
+    real_type momentum_inv = celeritas::rsqrt(dot_product(y.mom, y.mom));
 
-    // Evaluate the right-hand-side of the equation
+    // Evaluate the rate of change in particle's position per unit length: this
+    // is just the direction
     OdeState result;
+    for (size_type i = 0; i != 3; ++i)
```

Comment:
Could add a comment on why this is advantageous over the `aspy` call?

Reply:
It's not a matter of advantageous: `axpy` means `y <- a * x + y`, and the operation we're trying to do here is `y <- a * x`. Previously it was effectively `y <- a * x + 0` based on the fact that `y` was initialized to zero.

Reply:
Why not use a (new) `ax` routine? (for the same reason we have `axpy` :) ).

Reply:
Ha, it looks like there *is* an `ax` in FieldUtils just for this purpose... I've updated.